devsetup/Makefile

```diff
@@ -662,27 +677,42 @@ bmaas_metallb_cleanup:
 bmaas_virtual_bms: export NODE_COUNT = ${BMAAS_NODE_COUNT}
 bmaas_virtual_bms: export NETWORK_NAME = ${BMAAS_NETWORK_NAME}
 bmaas_virtual_bms: export NODE_NAME_PREFIX = ${BMAAS_INSTANCE_NAME_PREFIX}
+bmaas_virtual_bms: export NODE_NAME_SUFFIX = ${BMAAS_INSTANCE_NAME_SUFFIX}
 bmaas_virtual_bms: export MEMORY = ${BMAAS_INSTANCE_MEMORY}
 bmaas_virtual_bms: export VCPUS = ${BMAAS_INSTANCE_VCPUS}
 bmaas_virtual_bms: export DISK_SIZE = ${BMAAS_INSTANCE_DISK_SIZE}
 bmaas_virtual_bms: export OS_VARIANT = ${BMAAS_INSTANCE_OS_VARIANT}
 bmaas_virtual_bms: export VIRT_TYPE = ${BMAAS_INSTANCE_VIRT_TYPE}
 bmaas_virtual_bms: export NET_MODEL = ${BMAAS_INSTANCE_NET_MODEL}
 bmaas_virtual_bms: ## Create libvirt VM for BMaaS
-	scripts/bmaas/vbm-setup.sh --create
+	if [ ${NODE_COUNT} -eq 1 ]; then \
+		scripts/bmaas/vbm-setup.sh --create ; \
+	else \
+		for INDEX in $(shell seq 0 $$((${NODE_COUNT} -1))) ; do \
```

Comment:
Calling these scripts as many times as the number of nodes is possibly little inefficient, though does not matter probably as no one would be deploying large number of nodes with install_yamls.

Reply:
Why do we need to iterate in the Makefile though? The `vbm-setup.sh` takes a `NODE_COUNT` env var and iterates within the script itself:
https://github.com/openstack-k8s-operators/install_yamls/blob/main/devsetup/scripts/bmaas/vbm-setup.sh#L166-L170

https://github.com/openstack-k8s-operators/install_yamls/blob/main/devsetup/scripts/bmaas/vbm-setup.sh#L145-L147


```
if [ ${NODE_COUNT} -eq 1 ]; then \
 		scripts/bmaas/vbm-setup.sh --create ; \
 	else
 	        scripts/bmaas/vbm-setup.sh --create --num-nodes $NODE_COUNT
fi
```

Then we can just handle the `NODE_NAME_SUFFIX` in the script? 

Just avoids re-running the script like Rabi is mentioning/

Reply:
I guess the whole check isn't really needed in that scenario actually, because `NODE_COUNT` would just be handled in the script anyway.

Reply:
This PR changes vbm-setup.sh to drop --num-nodes, so it manages just a single VM. That change is already included in this PR. Are you asking me to drop that change from the PR? The only inefficient part of calling the script twice would be setting up the libvirt logging, not really something I'd optimize in a dev repo, but we can go back to the way it was if that is wanted.

gen-edpm-node.sh manages 1 vm at a time. That is used for networker and bootc vms already, with the Makefile controlling the looping. So, I was changing the baremetal scripts to work in a similar way.

gen-edpm-node-bgp.sh is hardcoded to manage 3 at a time, so that's a different way as well.

Whichever way it's done, I'd like to see these things consistent.

Reply:
@bshephar @rabi can you take another look at this one?

I made the requested changes so that the looping is still done in the script(s) now. I think this is an overall improvement given what it enables.

Reply:
@hjensas @steveb as well.